### ISSUE TITLE
O3 optimization are slower on SG2042

### ISSUE THREAD

**Issue by xhebox** [MegEngine/InferLLM#60]
Seems that O3 optimization is too aggressive on most RISCV systems, where pipeline/caching is not doing as well as x86. Disable O3 makes chatglm2 runs at 3.6t/s compared to 3t/s.

**Comment by chenqy4933** [comment]
its sames the compiler issue

**Comment by xhebox** [comment]
> its sames the compiler issue

Yep. But I am thinking disable O3 by default. Or make it a CMake option somehow.

**Comment by chenqy4933** [comment]
I agree it
